rfcs/configure-file-name-from-block-meta.md

```diff
@@ -0,0 +1,82 @@
+# RFC: Configuring File Name from Block Meta
+
+## Summary
+
+Allowing code blocks to change the `filename` used in `eslint-plugin-markdown`'s processor using codeblock `meta` text.
+
+## Motivation
+
+Some projects use ESLint `overrides` to run different lint rules on files based on their file name.
+There's no way to respect file name based `overrides` in parsed Markdown blocks right now using `eslint-plugin-markdown`'s parsing.
+This RFC would allow code blocks to specify a custom file name so that `overrides` can be used more idiomatically.
+
+### Real-World Example
+
+In [`create-typescript-app`](https://github.com/JoshuaKGoldberg/create-typescript-app), `*.json` files _except_ `package.json` files have [`jsonc/sort-keys`](https://ota-meshi.github.io/eslint-plugin-jsonc/rules/sort-keys.html) enabled using [an ESLint config file `overrides`](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/76a75186fd89fc3f66e4c1254c717c28d70afe0d/.eslintrc.cjs#L94).
+However, a code block in a file named `docs/FAQs.md` intended to be a `package.json` is currently given a name like `create-typescript-app/docs/FAQs.md/0_0.jsonc`.
+Short of hardcoding overrides or disabling the lint rule in the file, there is no way to have the `overrides` apply to the markdown code block.
+
+## Detailed Design
+
+This RFC proposes that code blocks be allowed to specify a file path in `meta` with `title="..."`.
+Doing so would replace the `filename` provided by `eslint-plugin-markdown`'s `preprocess` method.
+
+````md
+```json title="package.json"
+{}
+```
+````
+
+Parsing would be handled by a regular expression similar to the [Docusaurus parser](https://github.com/facebook/docusaurus/blob/7650829e913ec4bb1263d855719779f6b97066b6/packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts#L12).
+The `title` meta prop name is intentionally the same as Docusaurus, so projects can use the same attribute for Docusaurus and ESLint.
+
+Parsed language is ignored when the codeblock provides a custom title.
+Some languages have many file extensions, such as TypeScript's `.cts`, `.ts`, `.tsx`, etc.
+Some extensions may map to multiple languages, such as Perl and Prolog both using `.pl`.
+
+Roughly:
+
+```diff
+- filename: `${index}.${block.lang.trim().split(" ")[0]}`,
++ filename: titleFromMeta(block) ?? `${index}.${block.lang.trim().split(" ")[0]}`,
+```
+
+### Name Uniqueness
+
+Codeblocks must have unique file paths for ESLint processing.
+If a codeblock has the same file path as a previously processed codeblock, processing should throw an error.
```

Comment:
After some thinking, I don't think throwing an error is a good idea. This might be a case where the Markdown is showing how a file changes over time (I do this in my writing), in which case changing `title`/`filename` would not make sense for the reader.

Instead, what do you think about appending a number, so the first instance would be `filename.js` but the second would be `filename-2.js`?

Reply:
I like that! +1, makes sense to me.